### PR TITLE
simplifying installation, documentation and npm update

### PR DESCRIPTION
**Pull request by vinyll**
Step 1 of simplifying the installation:

- updated readme installation
- moved readme _rst_ to _md_ (planning to move the entire doc to md)
- moved _inv_ assets tasks to _npm_ (`inv assets` and `inv watch` to `npm run assets:build` and `npm run assets:watch`) while preserving inv aliases.
- updated npm babel dependencies to fixed latest
- fixed a broken `div` in udata/templates/header.html

> This was successfully tested on a brand new VirtualBox'ed Xubuntu 15.10 desktop

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+uData
+=====
+
+[![Build status](https://secure.travis-ci.org/etalab/udata.png)](https://coveralls.io/repos/etalab/udata/badge.png?branch=master)
+[![Code coverage](https://coveralls.io/repos/etalab/udata/badge.png?branch=master)](https://coveralls.io/repos/etalab/udata/badge.png?branch=master)
+[![Python Requirements Status](https://requires.io/github/etalab/udata/requirements.png?branch=master)](https://requires.io/github/etalab/udata/requirements/?branch=master)
+[![JavaScript Dependencies Status](https://david-dm.org/etalab/udata.svg)](https://david-dm.org/etalab/udata)
+[![Join the chat at https://gitter.im/etalab/udata](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/etalab/udata?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+
+uData is an open and social data hub. This currently a work in progress and API is subject to change until the first official release.
```
Comment (davidbgk) [inline README.md]: Maybe we should update that? @noirbizarre ? (btw, missing `is` I think)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+### Installing the project
+
+Let's call the root project `$WORKSPACE`.
+
+    mkdir $WORKSPACE && cd $_
+
+Retrieve the sources
+
+    git clone https://github.com/etalab/udata.git
+Install your [virtualenv](https://virtualenv.readthedocs.org/en/latest/)
```
Comment (davidbgk) [inline README.md]: Requires a newline?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+#### Installing natively
+
+If you really don't want to use docker you can install MongoDB 3.2, ElasticSearch 1.7 with analysis-icu plugin, and Redis 3.0 on your way.
+
+Now open your browser to http://localhost:7000/. It's empty and ugly, yet it works!
+
+
+### Building CSS
```
Comment (davidbgk) [inline README.md]: And JS?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+To build the CSS run in a dedicated terminal:
+
+    npm assets:watch
```
Comment (davidbgk) [inline README.md]: Is the `assets_watch` method automatically converted to `assets:watch`?

Comment (davidbgk) [inline README.md]: Oh wait I got it, it's now executed through `npm`.

File: udata/templates/header.html [hunk 5]
```diff
@@ -32,70 +32,69 @@
         {% endfor %}
     </ul>
 
-      <ul class="nav navbar-nav navbar-right collapse navbar-collapse">
-        {# <li><a href="{{ url_for('front.metrics') }}">{{ _('Metrics') }}</a></li> #}
-        {% if current_user.is_authenticated() %}
-        <li class="dropdown">
+    <ul class="nav navbar-nav navbar-right collapse navbar-collapse">
```
Comment (davidbgk) [inline udata/templates/header.html]: Is that related to that PR?

File: tasks.py [hunk 6]
```diff
@@ -178,29 +178,29 @@ def i18nc():
 
 
 @task
-def assets(progress=False):
+def assets_build(progress=False):
```
Comment (davidbgk) [inline tasks.py]: It needs to be updated on `udata-gouvfr` too for consistency.

**Comment by davidbgk** [comment]
That PR made my day!

File: README.md (hunk 4 above)
Comment (vinyll) [inline README.md]: mistake, it's `npm run assets:watch`.
`inv watch` calls `npm run assets:watch
`

File: udata/templates/header.html (hunk 5 above)
Comment (vinyll) [inline udata/templates/header.html]: The `div` was broken and I wasn't sure whether the sample dataset i'm preparing were not being displayed for that reason. It was a different reason in the end yet this was fixed.

File: tasks.py (hunk 6 above)
Comment (vinyll) [inline tasks.py]: I didn't have that in mind. Good reminder for next steps :)

File: README.md (hunk 3 above)
Comment (vinyll) [inline README.md]: I omitted a chapter. fixed!

**Comment by vinyll** [comment]
Glad it made your day, it made my week! :stuck_out_tongue_winking_eye:

File: README.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+Let's call the root project `$WORKSPACE`.
+
+    mkdir $WORKSPACE && cd $_
+
+Retrieve the sources
+
+    git clone https://github.com/etalab/udata.git
+
+Install your [virtualenv](https://virtualenv.readthedocs.org/en/latest/)
```
Comment (davidbgk) [inline README.md]: Is that really required if all we do is launching the docker-compose command? (Really curious because I don't use Docker)

File: README.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+Now open your browser to http://localhost:7000/. It's empty and ugly, yet it works!
+
+## JS dependencies
+
+You need NodeJS 4.2. If it's not already installed or you have a different version, you should consider [installing NVM](https://github.com/creationix/nvm#installation).
+
+
+Then install JavaScript dependencies
+
+    cd data
```
Comment (davidbgk) [inline README.md]: `data` or `udata`?

File: README.md (hunk 7 above)
Comment (vinyll) [inline README.md]: docker-compose will take care of elasticsearch, redis and mongodb. It does not bundle the python project itself.

Comment (noirbizarre) [inline README.md]: It depends on if we use docker-compose for the stack services only (Mongo, Elasticsearch, Redis...) or the development environment too. Right now, we are using docker-compose only for services, not the complete environment.

File: tasks.py [hunk 9]
```diff
@@ -178,29 +178,29 @@ def i18nc():
...
     '''Install and compile assets'''
     header('Building static assets')
-    cmd = 'webpack -c --config {0}.js'
+    cmd = 'npm run assets:build -- --config {0}.js'
     if progress:
         cmd += ' --progress'
     nrun(cmd.format('webpack.config.prod'), pty=True)
```
Comment (noirbizarre) [inline tasks.py]: I think `nrun` should be replaced by `lrun`. It's only purpose was to replace to transform from `nrun('my-cmd')` to `lrun('node_modules/.bin/my-cmd')`.

File: tasks.py [hunk 10]
```diff
@@ -178,29 +178,29 @@ def i18nc():
...
 @task
-def watch():
-    nrun('webpack -d -c --progress --watch', pty=True)
+def assets_watch():
+    nrun('npm run assets:watch', pty=True)
 
 
 @task
 def widgets():
     nrun('webpack -d -c --progress --watch --config webpack.widgets.config.js',
```
Comment (noirbizarre) [inline tasks.py]: This one should also be in `package.json` and then replaced here.

**Comment by vinyll** [comment]
The test broke because of readme.rst -> readme.md which was parsed for setup.py long-description.
I refactored setup.py accordingly

File: setup.py [hunk 11]
```diff
@@ -70,7 +62,7 @@ def dependency_links(filename):
     name='udata',
     version=__import__('udata').__version__,
     description=__import__('udata').__description__,
-    long_description=long_description,
+    long_description=clean_doc('README.md'),
```
Comment (noirbizarre) [inline setup.py]: The main problem is that long_description expected some restructuredtext :/

File: setup.py [hunk 12]
```diff
@@ -56,21 +57,15 @@ def dependency_links(filename):
...
 install_requires = pip('install.pip')
 tests_require = pip('test.pip')
+readme_md = clean_doc('README.md')
 
 setup(
     name='udata',
     version=__import__('udata').__version__,
     description=__import__('udata').__description__,
-    long_description=long_description,
+    long_description=pypandoc.convert(readme_md, 'rst', format='md'),
```
Comment (vinyll) [inline setup.py]: @noirbizarre: converting the readme _md_ to _rst_ description to match the pypi requirements.

Comment (davidbgk) [inline setup.py]: @vinyll overkill? http://stackoverflow.com/a/26737672

Comment (vinyll) [inline setup.py]: @davidbgk: The link you provide is the same technique with the same lib.
How would you consider simplifying?

Comment (davidbgk) [inline setup.py]: Huhu, I didn't noticed the change, nevermind ;)

Comment (noirbizarre) [inline setup.py]: It can be a solution but it makes `pandoc/pypandoc` a requirements :/

Another solution is to keep the README in restructured text but minimalist and move the content in the documentation index and or quickstart (it's mostly done in #381)

Comment (noirbizarre) [inline setup.py]: Or third solution: embed the long_description directly into the setup.py. It won't be extracted anymore from the README, but who cares, we don't publish on PyPI right now :)

Comment (davidbgk) [inline setup.py]: I'm in favor of just providing a link to http://udata.readthedocs.org/en/latest/

Comment (vinyll) [inline setup.py]: _pypandoc_ is a dependency only for the develop.pip file. Just like transiflex-client, honcho, or Flask-DebugToolbar (any of you use it BTW?).
I'd rather either keep the dependency or make a link as suggested.
